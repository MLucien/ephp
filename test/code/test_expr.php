<?php

class C { public $hi = "world"; }
$c = new C;

$a = [ "a" => 51  // index a 
     , "b" => 100 // index b
     , "c" => "this is a {$c-> hi }"
     , "d" => (12 + 24 + # this is a comment
               48 + 96 + /* another */
               192 + 384)
     ];
var_dump($a);
var_dump(1 xor 0);
var_dump(20 % 3);
var_dump(5 << 1);
var_dump(5 >> 1);

$a = 10;
$a += 10;
var_dump($a);
