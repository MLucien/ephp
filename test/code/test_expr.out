array(4) {
  ["a"]=>
  int(51)
  ["b"]=>
  int(100)
  ["c"]=>
  string(15) "this is a world"
  ["d"]=>
  int(756)
}
bool(true)
int(2)
int(10)
int(2)
int(20)
